versioned_docs/version-1.0/cadence-migration-guide/usdc-migration.mdx

```diff
@@ -0,0 +1,336 @@
+you need to do to migrate your usage of that app to the new `USDCFlow` token.
+The Flow foundation is already working with most apps that use USDC
+to help them with their migration plans to the new `USDCFlow`,
+so rest assured that there will be a path for you to retain your assets
+past the Crescendo migration, regardless of what form that takes.
+
+If you cannot get in touch with the support channels of whatever app you use,
+please reach out in the [Flow Discord](https://discord.com/invite/J6fFnh2xx6)
+with your issues and the Flow team will definitely help you get it resolved.
+
+## Flow Users with Marketplace listings for USDC
+
+If you are a user with a NFT listing on a marketplace with USDC as the payment currency,
+your existing listings will not work any more after the Crescendo upgrade.
+If you still want your listings to work, you will need to create new listings
+with the new `USDCFlow` currency so they are still available after the Crescendo upgrade.
+This will likely require that whatever app you were using for marketplace listings
+supports the transaction that lists for `USDCFlow` instead of `FiatToken`,
+so if this isn't immediately available, please contact their support
+to ask if it will be supported.
+Luckily, this is not time sensistive and can be completed
+any time after the Crescendo migration with no downside besides
+old listings not being purchasable until new ones with `USDCFlow` are created.
+
+## Flow Users with USDC in Defi
+
+If you are a user of USDC in a defi application, please reach out
+to the support channel for any application that you use to see if there
+is anything that you are able to do to migrate to the new `USDCFlow` token
+without needing to swap manually. Some projects, such as Increment.fi, are designing
+ways to migrate their users to the new `USDCFlow` token without users needing 
+to take any actions on their own, but these solutions will likely be announced
+by their respective projects.
+
+If you are using these projects and want to take the safest action,
+unless you hear otherwise from the project, your best bet is to try
+to unwind all your existing `FiatToken` and swap it to `USDCFlow` 
+so you can be ready when they eventually support `USDCFlow`.
+
+# USDC Migration Developer Guide
+
+If you are a developer who currently uses `FiatToken` in some way, you will likely
+have to do some work to remove the dependency and/or migrate
+your code and state to the new `USDCFlow` token contract.
+
+You will likely have to make multiple sets of changes if you want 
+this process to go as smoothly as possible. This obviously depends
+on the project and what kind of integration with USDC you already have.
+
+The first set is the pre-crescendo changes
+to allow users to pull their USDC out of your contracts, swap to the new `USDCFlow` token,
+and then re-enter your updated contract.
+Second, you'll have to update these changes to Cadence 1.0 so that they
+continue to work after Crescendo.
```

Comment:
```suggestion
It's necessary to complete the following steps **before** the Crescendo milestone:

1. To allow users to pull their USDC out of your contracts, swap to the new `USDCFlow` token,
and then re-enter your updated contract.
2. Update these changes in your Cadence 1.0 contract versions and stage the updated contract code so that they continue to work after Crescendo.
```

> then re-enter your updated contract.

Not sure what's meant by "re-enter" here